dhis-2/dhis-support/dhis-support-artemis/src/main/java/org/hisp/dhis/artemis/audit/listener/AbstractHibernateListener.java

```diff
@@ -102,63 +109,126 @@ public String getCreatedBy()
     abstract AuditType getAuditType();
 
     /**
-     * Try to initialize all lazy loaded collections of the given Entity before sending
-     * it to {@link AuditObjectFactory} for serializing to JSON
-     * @param factory {@link SessionFactoryImplementor}
-     * @param entity current Entity
-     * @return the Entity with all collections loaded
+     * Create Audit entry for update event
      */
-    protected Object initHibernateProxy( SessionFactoryImplementor factory, Object entity )
+    protected Object createAuditEntry( PostUpdateEvent postUpdateEvent )
     {
-        Session session = factory.getCurrentSession();
+        return createAuditEntry( postUpdateEvent.getEntity(), postUpdateEvent.getState(), postUpdateEvent.getSession(), postUpdateEvent.getId(), postUpdateEvent.getPersister() );
+    }
+
+    /**
+     * Create Audit entry for insert event
+     */
+    protected Object createAuditEntry( PostInsertEvent postInsertEvent )
```

Comment:
I would move "down" these two methods (`createAuditEntry`) to the sub-classes (`PostInsertAuditListener` and `PostUpdateAuditListener`)

Reply:
fixed